### DISCUSSION TITLE
remote dev support

### DISCUSSION THREAD

**Discussion by meicale** [LazyVim/LazyVim#480] [Q&A] [answered]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

As a dev, I have lovely dev-environment and different project. I want my dev-tools be stable and efficient with less change so we can have similar workflows. BUT, different projects have different dependencies.  My current solution is to separated them using different docker (Dockerfile) container. 

### Describe the solution you'd like

Lots of Neovimer come from VSCode which take advantage of the remote-dev plugins to make remote development quite easy. NeoVim has better local dev experience, but has much less support remotely.  My current config can use DAP to debug on remote container or server. But I cannot make LSP and test (such as neotest) can not work this way. 

### Describe alternatives you've considered

I tried distant.nvim and maybe we can figure out a better solution if you guys like that idea.  Feel free to give any ideas and comments!

### Additional context

Thanks a lot! May you happy!

**Comment by chipsenkbeil** [comment, marked as answer]
@meicale is there a reason `distant.nvim` doesn't meet your needs? It supports language servers among many other features on the v0.2 branch.

**Reply by meicale** [reply]
Debug(DAP), LSP, and unit test are all quite useful for developer. I expect better surport on these 3 things in nvim remote setups. I can not take distant.nvim as a solution:
1. distant.nvim should be cool and I tried it. Unfortunately, it doesn't work fine when using dev-tool and project container and I reported this as an open issue. 
2.  Just distant.nvim is not enough yet. It is better that run all the test in project container and have the results collected in the dev-tool container. I use neotest and it works fine locally and it is no easy way to run test "DAP" way.
I think dap and server like debug-py are best solution for debug python file so far in remote setting up, so the similar abililty is expect in  unit-test and lsp.
You are welcome to make more demo and setup video on your yt channel. @chipsenkbeil 
distant .nvim is useful and it is cooler  adding this plugin in lazyvim. @chipsenkbeil @folke

**Comment by emretuna** [comment]
https://github.com/esensar/nvim-dev-container
this would be a nice addition.
